### PR TITLE
[docs] Edit mixin READMEs and doc-gen script

### PR DESCRIPTION
**Pull request by jesscodez**
@tehut @TamikoT PTAL
@hausdorff fyi

TLDR:
* **Made edits to the `main.go` script in `doc-gen/`** (see the README in that subdirectory for details). Now you only need to specify the path to the mixin library when running the command (obv some baked in assumptions about the existence of a `mixin.yaml` file and optional `prototypes/` directory)
* **Convert `mixin.yaml` files to actually be YAML and not JSON** -- not sure if this is going to muck with the `ks describe` work...but YAML helps a lot with readability (esp with the `description` field)
* **Small edits to the README template** for readability/organization
* **Made standardizing edits to actual text** in the `mixin.yaml` and `prototypes/*.jsonnet` files

### CODE REVIEW

**Comment by jesscodez** [comment]
out of date, closing!
